Edit Details

mun9769 opened the pull request:
## 🔥 *Pull requests*

⛳️ **작업한 브랜치**
- #99 

👷 **작업한 내용**
- 광고배너 자동으로 넘어가기 
- 주민등록번호 생년월일 유효성 검사 구현 
- 로딩중일때 로딩 애니메이션 넣기 
- 창넘어가는 애니메이션 disable하기 (단, 모달창은 되야함) 
- MainView > 타이틀(로고) 크기 크게하기(커스텀) 
- PatientInfoView 전화번호 없어요를 눌렀을 때 다음버튼 비활성되는 버그 해결
- 정규표현식 적용해서 틀리면 textField이 경고를 낸다
- 틀렸을 경우 textfield를 만든다
- 스크롤 내려갈때 메인뷰 네비게이션 바는 고정
- 광고 움직이기
- 카카오와 파베auth 패키지 제외
- homenavigation만 dismiss 가능한 이슈 해결하기
- 파베에 문서 보내기 
- 버튼 애니메이션 적용( 천천히 색이 변함 )
- ~~모든 뷰에 keyboard 버튼을 적용하기~~

navigationStack에 관한 이슈에 대해 글을 작성했어요 나중에 의논해보면 좋을듯합니다
https://velog.io/@mun9769/EnvironmentObject%EC%9D%98-%EC%B6%94%EC%83%81%ED%99%94-%EB%AC%B8%EC%A0%9C

keyboard 버튼이 아직도 이슈가 있습니다 당장 해결할 수 없어서 해야할일 목록에 추가해두고 나중에 해결하겠습니다
[해야할일목록](https://github.com/orgs/DeveloperAcademy-POSTECH/projects/91/views/3?pane=issue&itemId=42367213)

firebase auth 패키지를 지우고 싶지만 firebase storage 기능을 사용해야해서 지울 수 없었습니다 
단, auth 패키지를 프로젝트에서 제외했으니 빌드 속도가 소폭 빨라졌을 거에요


## 📸 스크린샷
|기능|버튼누름효과|에러텍스트필드|
|:--:|:--:|:--:|
|기능|<img src = "https://github.com/DeveloperAcademy-POSTECH/MacC-Team2-Nutty/assets/59304977/1a194d92-ce77-45e6-913c-7d36e20073f7" width=120>|<img src = "https://github.com/DeveloperAcademy-POSTECH/MacC-Team2-Nutty/assets/59304977/2f017a72-596e-4645-8b81-dfbb4dfd5e68" width=120>|

mun9769 committed:
[Fix/#99] navigationview fix

navigationView를 상단에 고정시킨다

mun9769 committed:
[Feat/#99] Edit HomeView detail

패딩 조정, 글자크기 키움

mun9769 committed:
[Fix/#99] fix keyboard button

toolbar를 사용할 때 클릭되는 영역이 제한되는 이슈가 있어서 수정이 필요했음
ignoresSafeArea(.keyboard)를 적용하면 키보드와 버튼의 간격이 생기는 버그를 해결할 수 있음.

mun9769 committed:
[chore/#99] debug mode

debug마다 validate를 체크하는게 번거로워서 if처리함

mun9769 committed:
[Feat/#99] CTAButton animation

버튼이 눌렸을 때 배경색이 진해진다

mun9769 committed:
[Fix/#99] remove LoginView, kakao package

로그인을 사용하지 않기 때문에 LoginView와 AuthenticationViewModel은 주석처리한다
kakao package는 Onbom 프로젝트에서 지운다
firebase package를 지우고 싶지만, firebaseStorage 기능을 사용하므로 지우지 않았다.
단, firebaseAuth는 Onbom 타켓에서 지워주었다.

mun9769 committed:
[Feat/#99] upload FirebaseStorage

debug 모드마다 파일을 업로드하지 않는다
만약 업로드 과정에서 에러가 발생한다면 에러메세지를 보여주는 화면으로 전환한다

더 할일)
전송시간을 2.5초로 해도 괜찮은지 물어보기
SubmitLoadingView의 에러처리 코드를 줄일 수 있는지 고민해본다.

mun9769 committed:
[Feat/#99] throw error when fail to upload pdf

debug 모드일 땐 3초 딜레이 주기만 함.
error가 들어오면 view는 에러화면으로 전환된다

mun9769 committed:
[Fix/#99] Add wrong textfield case

release모드일 때만 validate하도록 전처리기를 넣음

mun9769 committed:
[chore/#99] add tabgesture

탭바가 눌린건지 아닌지 헷갈리기 때문에 탭뷰로 넘어갈 수 있도록 수정

mun9769 committed:
[fix/#99] fix validation

매 입력마다 validation하도록 수정
testcode에서 name이 valid한지 확인

mun9769 committed:
[chore/#99] edit detail

mun9769 committed:
Merge branch 'dev' into 99-edit-details

mun9769 committed:
[Setting/#99] merge issue

pbxproj 파일을 수정한다
머지할 때 놓친 수정부분을 고친다
xcode에서 빌드할 때 잘 동작하는 것을 확인했다

mun9769 committed:
[Fix/#99] edit color FormTextField

mun9769 committed:
[Fix/#99] fix completion

3초 뒤에 completion이 호출된다 -> 다음뷰로 넘어간다

ge-um commented on Onbom/Onbom/View/HomeView.swift:
```diff
@@ -15,153 +15,153 @@ struct HomeView: View {
-                .onReceive(timer, perform: { _ in
-                    withAnimation {
-                        selectedPage = (selectedPage + 1) % 2
+        VStack(spacing: 0){
```
P5: ㅋㅋ자잘한 거긴 한데 (spacing: 0) 뒤에 띄어쓰기 부탁드립니다

mun9769 replied:
노력해볼게요. 나중에 lint 적용하면 좋을듯 싶어요

ge-um commented on Onbom/Onbom/View/HomeView.swift:
```diff
@@ -15,153 +15,153 @@ struct HomeView: View {
+                    .foregroundColor(Color.PB4)
+                    .font(.custom("Dongle-Bold", size: 38))
+                    .padding(.leading, 20)
+            } confirmation: {
```
P2: 여기 코드가 있진 않은데 CustomNavigationBar가 지금 HomeView에서만 쓰이지 않나요?  HomeView의 CustomNavigationBar에는 확인과 취소 기능이 들어가있지 않은데 parameter로 cancellation과 confirmation이라는 이름을 쓴 게 왜인지 궁금해요 그리고 둘다 view인데 왜 V랑 R로 이름 지으신건가요? 궁금쓰

mun9769 replied:
CustomNavigation은 충분히 이름이 바꿀 수 있어요. 다른뷰에 사용될 가능성이 있어서 Custom이란 prefix를 붙였어요
cancellation과 confirmation보다 위치를 나타내는 네이밍이 더 적절한 것같아서 leading과 trailing으로 수정했어요
V는 View고 R은 Right의 약자입니다~

SeBin-Kwon commented on Onbom/Onbom/View/LTCI/SubmitLoadingView.swift:
```diff
@@ -8,29 +8,65 @@
-        .onAppear() {
-            DispatchQueue.main.asyncAfter(deadline: .now() + 1.5) {
-                isSubmitViewPresented = true
+        else if (self.state == .fail) {
+            Text(self.err ?? "에러")
+                .navigationBarBackButton()
+        }
```
P5: '다시 시도해주세요' 라는 메시지로 바뀐다면 다시 시도할 수 있는 버튼이 필요할 것 같네요ㅎㅎ

mun9769 replied:
수정했습니다~

SeBin-Kwon reviewed (approved):
수고하셨습니다~ 다음엔 꼭 이슈 나눠서 해주세요ㅎㅎ 에렌이 어떤걸 하고 계시는지 하나하나 알기가 힘들어서 작업이 겹칠 수 있으니까여...

ge-um commented on Onbom/Onbom/View/HomeView.swift:
```diff
@@ -15,153 +15,153 @@ struct HomeView: View {
-                            .padding(.trailing, i == 5 ? 20 : 10)
                         }
                     }
+                    Spacer().frame(height: 100)
```
P5: 에이 .frame 띄워줘요~~

mun9769 replied:
싱글라인을 선호합니다 (찡긋)

ge-um commented on Onbom/Onbom/View/PatientInfoView.swift:
```diff
@@ -21,7 +21,7 @@ struct PatientInfoView: View {
+    @State private var isKeyboardVisible:       Bool = true
     @State private var isPressed:               Bool = false
     
     @ObservedObject private var viewModel = PatientInfoViewModel()
```
P4: 에이 이거 그냥 viewModel이라고 하지말고 patientInfoViewModel이라고 제대로 써줘요~~

ge-um commented on Onbom/Onbom/Core/ViewModel/PatientInfoViewModel.swift:
```diff
@@ -42,7 +42,11 @@ final class PatientInfoViewModel: ObservableObject {
     private var IDNumber1Publisher: AnyPublisher<Bool, Never> {
         $seniorIDNumber1
             .map { number in
+                #if DEBUG
```
P5: 이 코드 잊지 말고 나중에 출시 할 때는 지워줘요~

mun9769 replied:
release할 때는 전처리기가 DEBUG문을 코드에서 제외시켜줍니다~

ge-um commented on Onbom/Onbom/Components/FormTextField.swift:
```diff
@@ -10,33 +10,41 @@ import SwiftUI
+            ZStack(alignment: .trailing) {
+                TextField(placeHolder, text: $textInput)
+                    .font(.custom("Pretendard-Medium", size: 16))
+                    .lineSpacing(16 / 2 * (100 - 100)/100)
```
P5: 띄어쓰기하셈.........

mun9769 replied:
이것도 swiftlint를 사용함이 좋을듯해요

ge-um commented on Onbom/Onbom/View/PatientInfoView.swift:
```diff
@@ -45,32 +44,38 @@ struct PatientInfoView: View {
+                                    .keyboardType(.numberPad)
+                                    .padding(16)
+                                    .background(RoundedRectangle(cornerRadius: 10)
+                                        .stroke(viewModel.isSeniorIDNumber1Wrong ? Color.R : focusedField == .seniorIDNumber1 ? Color.PB4 : Color.PB1, lineWidth: 1.5))
```
P5: 정렬 ㄱㄱ

mun9769 replied:
정렬하였습니다~

ge-um commented on Onbom/Onbom/View/PatientInfoView.swift:
```diff
@@ -45,32 +44,38 @@ struct PatientInfoView: View {
                     VStack(spacing: 8) {
                         Text("주민번호")
-                            .foregroundColor(focusedField == .seniorIDNumber1 || focusedField == .seniorIDNumber2 ? Color.PB4 : Color.G6)
+                            .foregroundColor(viewModel.isSeniorIDNumber1Wrong ? Color.R : focusedField == .seniorIDNumber1 || focusedField == .seniorIDNumber2 ? Color.PB4 : Color.G6)
```
P5 : 줄바꿈이 최선일듯...

mun9769 replied:
줄바꿈이 맞다고 생각해서 ? 기준으로 줄바꿈해주었습니다

ge-um commented on Onbom/Onbom/Core/ViewModel/PatientInfoViewModel.swift:
```diff
@@ -38,6 +44,19 @@ final class PatientInfoViewModel: ObservableObject {
+    }
+    
     
     private var IDNumber1Publisher: AnyPublisher<Bool, Never> {
```
P5: 1/2 말고 다른 이름을 고민해 보심이...

mun9769 replied:
시간되면은 좋은 이름 추천부탁드려요

ge-um reviewed (approved):
수고하셨습니다~

mun9769 committed:
[chore/#99] edit details

CustomNavigationBar에서 행위보단 위치가 적절한 네이밍같아서 leading과 trailing으로 변경하였다
삼항연산자에 대해 보기 좋게 줄바꿈을 해주었다
